workers/unity/Packages/io.improbable.gdk.core/Commands/CommandRequestId.cs

```diff
@@ -0,0 +1,124 @@
+using System;
+
+namespace Improbable.Gdk.Core.Commands
+{
+    public readonly struct CommandRequestId : IEquatable<CommandRequestId>, IComparable<CommandRequestId>
+    {
+        public readonly long Raw;
+
+        public CommandRequestId(long requestId)
+        {
+            Raw = requestId;
+        }
+
+        public static bool operator ==(CommandRequestId left, CommandRequestId right)
+        {
+            return left.Equals(right);
+        }
+
+        public static bool operator !=(CommandRequestId left, CommandRequestId right)
+        {
+            return !left.Equals(right);
+        }
+
+        public static bool operator <(CommandRequestId left, CommandRequestId right)
+        {
+            return left.CompareTo(right) < 0;
+        }
+
+        public static bool operator >(CommandRequestId left, CommandRequestId right)
+        {
+            return left.CompareTo(right) > 0;
+        }
+
+        public static bool operator <=(CommandRequestId left, CommandRequestId right)
+        {
+            return left.CompareTo(right) <= 0;
+        }
+
+        public static bool operator >=(CommandRequestId left, CommandRequestId right)
+        {
+            return left.CompareTo(right) >= 0;
+        }
+
+        public bool Equals(CommandRequestId other)
+        {
+            return Raw == other.Raw;
+        }
+
+        public override bool Equals(object obj)
+        {
+            return obj is CommandRequestId other && Equals(other);
+        }
+
+        public override int GetHashCode()
+        {
+            return Raw.GetHashCode();
+        }
+
+        public int CompareTo(CommandRequestId other)
+        {
+            return Raw.CompareTo(other.Raw);
+        }
+    }
+
+    public readonly struct InternalCommandRequestId : IEquatable<InternalCommandRequestId>, IComparable<InternalCommandRequestId>
```

Comment:
What is the difference between `InternalCommandRequestId` and `CommandRequestId`? Is it that they're two different wrappers for two different types of commands and could theoretically deviate from each other? Or are you trying to create type safety?

Reply:
They are the same data wise, but 1 is an id the GDK generates when you schedule a command to be send. And the internal ID is the ID the worker SDK returns when the command is actually send over the network.
Using 2 different types here ensures they aren't accidentally mixed up,